.template/addons/github/.github/workflows/review_code.yml.tt

```diff
@@ -1,58 +1,81 @@
 name: Review code
 
-on:
-  pull_request:
-    types: [opened, reopened, synchronize]
+on: pull_request
 
 env:
   DANGER_GITHUB_API_TOKEN: ${{ github.token }}
 
 jobs:
+  wait_for_test_build:
+    name: Wait for test build
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Wait for tests build
+        uses: lewagon/wait-on-check-action@v1.1.2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          check-name: 'Build Docker image'
+          running-workflow-name: 'Wait for test build'
+          allowed-conclusions: success
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          wait-interval: 20
+
   automated_code_review:
     name: Run Danger
+    needs: wait_for_test_build
     runs-on: ubuntu-latest
+    timeout-minutes: 5
     steps:
-      - uses: actions/checkout@v2.3.4
+      - name: Wait for tests to complete
+        uses: lewagon/wait-on-check-action@v1.1.2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          check-name: 'Unit tests'
+          running-workflow-name: 'Run Danger'
+          allowed-conclusions: success
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          wait-interval: 20
+
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
+      - name: Download tests coverage artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: test.yml
+          workflow_conclusion: success
+          name: coverage
+          path: coverage
+
       - name: Setup Ruby
         uses: ruby/setup-ruby@v1
         with:
           bundler-cache: true
 
-      <%- if WEB_VARIANT -%>
       - name: Setup Node
-        uses: actions/setup-node@v2
-        with:
-          node-version: '<%= NODE_VERSION %>'
-
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-
-      - name: Cache Yarn
-        uses: actions/cache@v2
-        id: yarn-cache
+        uses: actions/setup-node@v3
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          node-version: "16.16.0"
```

Comment:
should we use the `NODE_VERSION` instead of hardcode the version here @carryall 🤔

Reply:
Updated in [6a0d027](https://github.com/nimblehq/rails-templates/pull/351/commits/6a0d027d9500d065d9b71a736d02cae512342db8)